tutorials/vue/vue-sfc.md

```diff
@@ -0,0 +1,76 @@
+Vue global components are great for small to medium sized projects but if your project becomes more complex,
+problems begin to arise like:
+
+- Every component name must be unique
+- No CSS support in the components
+- Restricted to HTML and JavaScript, instead of preprocessors like Pug or Babel
+- String templates lack syntax highlighting as well as requiring slashes for multiline HTML
+
+However, Vue.js's single-file components solve all these problems! A single-file component has the extension `.vue`,
+which is made possible by a module bundler like Webpack or Browserify. A `.vue` file can have many top-level language blocks
+as well as your own custom blocks, but the three core blocks are `<template>`, `<script>`, and `<style>`.
+When making a component, you do not need all three of these in the `.vue` file so you can omit a language block if
```

Comment:
You can make this sentence more concise by just saying "The 3 core blocks are all optional."